connector/core/core-service/src/main/java/org/oaebudt/edc/core/OaebudtCoreExtension.java

```diff
@@ -1,23 +1,60 @@
 package org.oaebudt.edc.core;
 
 import org.eclipse.edc.runtime.metamodel.annotation.Extension;
+import org.eclipse.edc.runtime.metamodel.annotation.Inject;
 import org.eclipse.edc.runtime.metamodel.annotation.Provider;
 import org.eclipse.edc.spi.system.ServiceExtension;
+import org.eclipse.edc.spi.system.ServiceExtensionContext;
+import org.eclipse.edc.spi.types.TypeManager;
+import org.eclipse.edc.sql.QueryExecutor;
+import org.eclipse.edc.transaction.datasource.spi.DataSourceRegistry;
+import org.eclipse.edc.transaction.spi.TransactionContext;
 import org.oaebudt.edc.core.store.InMemoryParticipantGroupStore;
+import org.oaebudt.edc.core.store.SqlParticipantGroupStore;
 import org.oaebudt.edc.spi.store.ParticipantGroupStore;
 
+import static org.eclipse.edc.transaction.datasource.spi.DataSourceRegistry.DEFAULT_DATASOURCE;
+
 @Extension(OaebudtCoreExtension.NAME)
 public class OaebudtCoreExtension implements ServiceExtension {
     public static final String NAME = "Core Extension";
 
+    @Inject
+    private DataSourceRegistry dataSourceRegistry;
+
+    @Inject
+    private TransactionContext transactionContext;
+
+    @Inject
+    private TypeManager typeManager;
+
+    @Inject
+    private QueryExecutor queryExecutor;
+
+    private ParticipantGroupStore store;
+
     @Override
     public String name() {
         return NAME;
     }
 
+    @Override
+    public void initialize(ServiceExtensionContext context) {
+        ServiceExtension.super.initialize(context);
+    }
+
     @Provider(isDefault = true)
-    public ParticipantGroupStore defaultParticipantStore() {
+    public ParticipantGroupStore inMemoryParticipantStore() {
         return new InMemoryParticipantGroupStore();
     }
 
+    @Provider
+    public ParticipantGroupStore createSqlDidEntryStore() {
+        if (store == null) {
```

Comment:
the best pattern to do this is to provide the sql store in a separate extension, so you can easily create different runtimes with different persistence methods (if we don't need to have both we can think about ditch the in-memory one)

Reply:
Hi @ndr-brt 

I remove both the InMemory and Mongo Impl. They are both not needed. Makes me wonder if creating a different extension still work in this context. 
What do you think?